tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py

```diff
@@ -0,0 +1,238 @@
+from typing import Set
+
+import pytest
+
+import great_expectations.exceptions.exceptions as ge_exceptions
+from great_expectations.data_context import DataContext
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.rule_based_profiler.parameter_builder import (
+    RegexPatternStringParameterBuilder,
+    regex_pattern_string_parameter_builder,
+)
+from great_expectations.rule_based_profiler.types import (
+    Domain,
+    ParameterContainer,
+    get_parameter_value_by_fully_qualified_parameter_name,
+)
+from tests.integration.profiling.rule_based_profilers.conftest import (
+    alice_columnar_table_single_batch,
+    alice_columnar_table_single_batch_context,
+    bobby_columnar_table_multi_batch_deterministic_data_context,
+)
+
+
+def test_regex_pattern_string_parameter_builder_instantiation():
+    candidate_regexes: Set[str] = {
+        r"^\d{1}$",
+        r"^\d{2}$",
+        r"^\S{8}-\S{4}-\S{4}-\S{4}-\S{12}$",
+    }
+
+    regex_pattern_string_parameter: RegexPatternStringParameterBuilder = (
+        RegexPatternStringParameterBuilder(
+            name="my_simple_regex_string_parameter_builder",
+        )
+    )
+
+    assert regex_pattern_string_parameter._threshold == 1.0
+    assert regex_pattern_string_parameter._candidate_strings == candidate_regexes
+    assert regex_pattern_string_parameter.CANDIDATE_STRINGS == candidate_regexes
+
+
+def test_regex_pattern_string_parameter_builder_zero_batch_id_error():
+    regex_pattern_string_parameter: RegexPatternStringParameterBuilder = (
+        RegexPatternStringParameterBuilder(
+            name="my_simple_regex_string_parameter_builder",
+        )
+    )
+    parameter_container: ParameterContainer = ParameterContainer(parameter_nodes=None)
+    domain: Domain = Domain(domain_type=MetricDomainTypes.COLUMN)
+
+    with pytest.raises(ge_exceptions.ProfilerExecutionError) as e:
+        regex_pattern_string_parameter.build_parameters(
+            parameter_container=parameter_container, domain=domain
+        )
+
+    assert (
+        str(e.value)
+        == "Utilizing a RegexPatternStringParameterBuilder requires a non-empty list of batch identifiers."
+    )
+
+
+def test_regex_pattern_string_parameter_builder_alice(
+    alice_columnar_table_single_batch_context,
+):
+    data_context: DataContext = alice_columnar_table_single_batch_context
+    batch_request = {
+        "datasource_name": "alice_columnar_table_single_batch_datasource",
+        "data_connector_name": "alice_columnar_table_single_batch_data_connector",
+        "data_asset_name": "alice_columnar_table_single_batch_data_asset",
+    }
+
+    candidate_strings: Set[str] = {
```

Comment:
Is it safe to say that this test and the remaining ones below are integration tests? The read from disk and perform a number of different assertions.

If we want to make these unit tests, let's maybe see how we can alter their scope a bit to isolate what exactly we're testing.

Reply:
@alexsherstinsky do we want to enforce our code standards in tests? If so, I think type annotations at the signature level are relevant too.

Reply:
Thank you for this discussion as well. Have simplified the assertion and added the type hint for `batch_request` (for consistency with other tests)